### PR TITLE
Add ContactInformation, PhoneNumber, Email, ContactLink components. Add Coordinators to ResourceDetail.

### PR DESCRIPTION
**Pull request by dtfreemn**
`ContactInformation` allows resources/events to have both an email address and a phone number and is extensible if we want to add other capabilities in the future. It routes emails to the `EmailAddress` component and phone numbers to the `PhoneNumber` component, both of which should be reusable. Those components pass their data to a final `ContactLink` component which handles the shared responsibility of trying to open the uri to actually initiate the communication. 

### **Resource with clickable phone number and email and Coordinators list if present**
image

### **Resource without contact info (renders normally):**
image

### CODE REVIEW

File: mobile/app/components/ContactInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    });
+  } catch (e) {
+    formattedContactInformation = null;
+  }
+
+  return formattedContactInformation && (
+    <View>
+      <View style={styles.contactContainer}>
+        <Icon source={icons.mail}/>
+        <Text style={styles.contactInformation}> Contact </Text>
```
Comment (kevinhylant) [inline mobile/app/components/ContactInformation.tsx:40]: Rm extra spaces around ` Contact ` - you can apply `paddingHorizontal: token.spacingXs` in `styles.contactInformation` instead

File: mobile/app/features/resources/ResourceDetail.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ export const ResourceDetail: React.FunctionComponent<Props> = ({
   },
 }) => {
   const coverImages = resource.get(Field.coverImages);
+  const contactInfo = resource.get(Field.contact);
```
Comment (kevinhylant) [inline mobile/app/features/resources/ResourceDetail.tsx:27]: We're going to have to rethink the data side of this. We definitely cannot rely on whoever ends up administrating this to input data as arrays & objects - they will definitely make a mistake & break our app.

Maybe something like `contactNumber` and `contactEmail`...?

Alternatively, and perhaps better would be to use a record in the `People` table to populate the contact info for an Event or Resource. There may be cases when we'd want to supply the location's general contact information, but I think we can sort that out in the code rather than in the data.
- People records already have a phone number & email so we can just grab those. Happy to pair with you on this!

File: mobile/ios/Podfile.lock [hunk 3]
```diff
@@ -26,11 +26,11 @@ PODS:
   - boost-for-react-native (1.63.0)
   - CocoaAsyncSocket (7.6.4)
   - CocoaLibEvent (1.0.0)
-  - CodePush (6.2.0):
+  - CodePush (6.3.0):
```
Comment (kevinhylant) [inline mobile/ios/Podfile.lock:29]: I'd avoid upgrading our packages as much as possible in PRs where you're adding features & keep upgrades separate

File: mobile/ios/Podfile.lock [hunk 4]
```diff
@@ -428,7 +428,7 @@ PODS:
     - React
   - RNReanimated (1.8.0):
     - React
-  - RNScreens (2.7.0):
+  - RNScreens (2.9.0):
```
Comment (kevinhylant) [inline mobile/ios/Podfile.lock:431]: Can you revert the changes to this file, Snyk automatically upgrades our pages in PR's for us & your changes here are likely to cause merge conflicts

File: mobile/app/components/PhoneNumber.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          onPress: (): void => console.log('Call cancelled.'),
+        },
+      ],
+      {cancellable: false},
+    );
+  };
+
+  return (
+    <View>
+      <Text
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:62]: Let's use the `Link` component I created for this & we don't need the extra `View` that wraps it

File: mobile/app/components/PhoneNumber.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      ? `telprompt:${phoneNumberNumsOnly}`
+      : `tel:${phoneNumberNumsOnly}`
+    : null;
+
+  async function makePhoneCall(): Promise<void> {
+    const showCallFailedAlert = () => Alert.alert('Oh no! That phone call didn\'t work');
+    
+    try {
+      // FYI -- doesn't work on ios emulator. will always error in local testing
+      const canCall = await Linking.canOpenURL(formattedPhoneNumber);
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:34]: I can show you how to build it on your phone to test this out before we merge it. I'll need some your phone's UUID in order to permit you to build it to your device, but we can sort that out

File: mobile/app/components/PhoneNumber.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  async function handlePressToCall(): Promise<void> {
+    Alert.alert(
+      `Call ${phoneNumberNumsOnly}`,
+      'Do you want to call this number?',
+      [
+        {
+          text: 'Yes',
+          onPress: makePhoneCall
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:49]: I **think**, that `Linking.openURL('tel:${phoneNumber}')` (<< pseudocode) will show the alert for us so we won't need this code I don't think!

File: mobile/app/components/PhoneNumber.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+}) => {
+  const phoneNumberNumsOnly = phoneNumber?.replace(/[^\d]/g, '');
+  const formattedPhoneNumber = (phoneNumberNumsOnly && phoneNumberNumsOnly.length >= 10)
+    ? Platform.OS
+      ? `telprompt:${phoneNumberNumsOnly}`
+      : `tel:${phoneNumberNumsOnly}`
+    : null;
+
+  async function makePhoneCall(): Promise<void> {
+    const showCallFailedAlert = () => Alert.alert('Oh no! That phone call didn\'t work');
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:30]: you can use `’` instead of `\'`  (shortcut key for the character is `opt + shift + ]`)

File: mobile/app/components/PhoneNumber.tsx (hunk 6 above)
Comment (kevinhylant) [reply mobile/app/components/PhoneNumber.tsx:34]: I think we can remove the `canCall ` bit entirely. If we try to do `Linking.openUrl` and it fails, the catch block will show an alert for us anyhow

File: mobile/app/components/PhoneNumber.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    
+    try {
+      // FYI -- doesn't work on ios emulator. will always error in local testing
+      const canCall = await Linking.canOpenURL(formattedPhoneNumber);
+      return canCall ? Linking.openURL(formattedPhoneNumber) : showCallFailedAlert();
+    } catch (e) {
+      console.log(e);
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:37]: rm

File: mobile/app/components/PhoneNumber.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    marginVertical: token.spacingSmallPlus,
+  },
+});
+
+export const PhoneNumber: React.FunctionComponent<Props> = ({
+  label,
+  phoneNumber,
+}) => {
+  const phoneNumberNumsOnly = phoneNumber?.replace(/[^\d]/g, '');
+  const formattedPhoneNumber = (phoneNumberNumsOnly && phoneNumberNumsOnly.length >= 10)
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:23]: Since we're going to end up using Airtable's `phoneNumber` field type from the People table we won't need to format the number as you're doing here

If we'd like to display it nicely, there's a good package for that `libphonenumber-js` << I think it is. I've used it before & it works nicely.

File: mobile/app/components/PhoneNumber.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+import { fonts, token } from 'assets';
+
+interface Props {
+  label: string;
+  phoneNumber: string;
+}
+
+export const styles = StyleSheet.create({
+  phoneNumber: {
+    ...fonts.regular,
```
Comment (kevinhylant) [inline mobile/app/components/PhoneNumber.tsx:12]: Let's rely on `Link`'s styling to display the link consistently to other links to make it clear that it is tappable using green rather than bolding. I know you were matching the mocks but in this case I think we should ignore her designs & be consistent about tappable items

File: mobile/app/components/ContactInformation.tsx (hunk 1 above)
Comment (dtfreemn) [reply mobile/app/components/ContactInformation.tsx:40]: yep. makes sense. done.

File: mobile/app/features/resources/ResourceDetail.tsx (hunk 2 above)
Comment (dtfreemn) [reply mobile/app/features/resources/ResourceDetail.tsx:27]: Yea. I agree with this in principle, just wasn't sure how to implement it in an extensible way. Having a `contactNumber` or `contactEmail` column would limit each resource to only having one of each, unless we were able to store them as arrays using Airtable's record linking functionality you mentioned. But, even in that case, there is also the issue of what is visually rendered vs the actual number/email for each contactNumber/Email. How would we account for that? I kinda figured that storing it as JSON was a bad idea, but I did it anyway as a first pass, assuming that we would have to figure out a better way.

File: mobile/app/components/PhoneNumber.tsx (hunk 6 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx:34]: done.

File: mobile/app/components/PhoneNumber.tsx (hunk 7 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx:49]: It didn't in my initial testing. So, maybe I wasn't passing some argument that I should, but there was no prompt without the manual confirmation alert.

File: mobile/app/components/PhoneNumber.tsx (hunk 8 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx:30]: done.

File: mobile/app/components/PhoneNumber.tsx (hunk 7 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx]: removed prompt for now. easy to add back in later if we want.

File: mobile/app/components/PhoneNumber.tsx (hunk 10 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx]: done.

File: mobile/app/components/PhoneNumber.tsx (hunk 11 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx]: done.

File: mobile/app/components/PhoneNumber.tsx (hunk 9 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx]: done.

File: mobile/app/components/PhoneNumber.tsx (hunk 5 above)
Comment (dtfreemn) [reply mobile/app/components/PhoneNumber.tsx]: done.

File: mobile/app/components/ContactInformation.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  email?: string;
+  phoneNumber?: string;
+};
+
+export const ContactInformation: React.FunctionComponent<Props> = ({
+  email,
+  phoneNumber
+}) => {
+  return (email || phoneNumber) && (
+    <View>
```
Comment (kevinhylant) [inline mobile/app/components/ContactInformation.tsx:20]: nit: (non-blocking) this is a good example of when you can use a fragment
`<>` instead of the View

File: mobile/app/components/ContactInformation.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+      <View style={styles.contactContainer}>
+        <Icon source={icons.mail}/>
+        <Text style={styles.contactInformation}>Contact</Text>
+      </View>
+      <View>
```
Comment (kevinhylant) [inline mobile/app/components/ContactInformation.tsx:25]: nit: by default elements in RN display vertically so this View is very likely not necessary

File: mobile/app/components/ContactInformation.tsx [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+export const ContactInformation: React.FunctionComponent<Props> = ({
+  email,
+  phoneNumber
+}) => {
+  return (email || phoneNumber) && (
+    <View>
+      <View style={styles.contactContainer}>
+        <Icon source={icons.mail}/>
+        <Text style={styles.contactInformation}>Contact</Text>
```
Comment (kevinhylant) [inline mobile/app/components/ContactInformation.tsx:23]: change: Can you use the `Message` component instead?

I'm trying to have us funnel 100% of all of our messages through it so that we can more easily translate things in the future & centrally apply styling

File: mobile/app/components/ContactLink.tsx [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  contactUri: string;
+};
+
+export const ContactLink: React.FunctionComponent<Props> = ({
+  contactDetail,
+  contactUri,
+}) => {
+  async function handlePress(contactUri: string): Promise<void> {
+    try {
+      // doesn't work on ios emulator. will always error in local testing
```
Comment (kevinhylant) [inline mobile/app/components/ContactLink.tsx:16]:
```suggestion
      // doesn't work on iOS simulator. will always error in local testing
```

File: mobile/app/components/ContactLink.tsx [hunk 16]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Link } from './Link';
+import { Linking, Alert } from 'react-native';
+
+interface Props {
+  contactDetail: string;
+  contactUri: string;
+};
+
+export const ContactLink: React.FunctionComponent<Props> = ({
```
Comment (kevinhylant) [inline mobile/app/components/ContactLink.tsx:10]: nit: Not sure it's worth having a separate component for this now that it's so stripped-down.. might just make it harder to quickly follow the code

File: mobile/app/components/ContactLink.tsx [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+      return Alert.alert('Oh no! That didn’t work');
+    }    
+  };
+  
+  return (
+    <Link onPress={handlePress}>
+      {contactDetail}
+    </Link>
+  );
+};
```
Comment (kevinhylant) [inline mobile/app/components/ContactLink.tsx:28]: just a small nit that Prettier normally picks up for you, but we generally include newlines at the end of our files... probably doesn't matter, but the idea is that it helps prevent errors with magnification or something 🤔

File: mobile/app/components/ContactLink.tsx [hunk 18]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Link } from './Link';
+import { Linking, Alert } from 'react-native';
+
+interface Props {
+  contactDetail: string;
```
Comment (kevinhylant) [inline mobile/app/components/ContactLink.tsx:6]: While you may end up getting rid of this component, might be more intuitive to just pass in the texts of the link as `children`

```<ContactLink uri={blahUri}>248-885-4004</ContactLink>```

File: mobile/app/features/resources/ResourceDetail.styles.tsx [hunk 19]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
 
-import { token, fonts } from '../../../assets';
+import { token, fonts } from 'assets';
```
Comment (kevinhylant) [inline mobile/app/features/resources/ResourceDetail.styles.tsx:3]: thanks!

You may have noticed that sometimes our automatic importing doesn't quite work as expected... it'll try to import some files from the `app` root... `app/components/common/Screen` but our Babel configuration isn't set up properly for that... it's on my to do list. Just wanted to call it out

File: mobile/app/features/resources/ResourceDetail.tsx [hunk 20]
```diff
@@ -41,10 +45,11 @@ export const ResourceDetail: React.FunctionComponent<Props> = ({
           <Message markdown style={styles.description}>
             {resource.get(ResourceField.description)}
           </Message>
+          <ContactInformation email={contactEmail} phoneNumber={contactPhoneNumber}/>
           <DetailPeople
             peopleIds={coordinatorIds}
             navigation={navigation}
-            subheader="Attendees"
+            subheader="Coordinators"
```
Comment (kevinhylant) [inline mobile/app/features/resources/ResourceDetail.tsx:52]: Thanks

File: mobile/app/components/ContactInformation.tsx (hunk 12 above)
Comment (dtfreemn) [reply mobile/app/components/ContactInformation.tsx:20]: yep. done.

File: mobile/app/components/ContactInformation.tsx (hunk 13 above)
Comment (dtfreemn) [reply mobile/app/components/ContactInformation.tsx:25]: yep. done.

File: mobile/app/components/ContactInformation.tsx (hunk 14 above)
Comment (dtfreemn) [reply mobile/app/components/ContactInformation.tsx:23]: done.

File: mobile/app/components/ContactLink.tsx (hunk 15 above)
Comment (dtfreemn) [reply mobile/app/components/ContactLink.tsx:16]: done.

File: mobile/app/components/ContactLink.tsx (hunk 17 above)
Comment (dtfreemn) [reply mobile/app/components/ContactLink.tsx:28]: done.

File: mobile/app/components/ContactLink.tsx (hunk 18 above)
Comment (dtfreemn) [reply mobile/app/components/ContactLink.tsx:6]: yep. done.

File: mobile/app/components/ContactLink.tsx (hunk 16 above)
Comment (dtfreemn) [reply mobile/app/components/ContactLink.tsx:10]: yea. i wondered about that, since it's so simple. but i like the fact that it maintains the single responsibility of actually opening the link and handling the error and the Email and PhoneNumber components also maintain a single responsibility of routing the appropriate url to that component. i'd like to keep it for now, if possible.

Comment (kevinhylant) [reply mobile/app/components/ContactLink.tsx:10]: But to that point `Link` could also centrally manage the try catching on a more global level?
